components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java

```diff
@@ -1309,6 +1346,251 @@ public void addApplicationWithTemplateIdAndTemplateVersionData(String templateId
                 REGISTRY_SYSTEM_USERNAME);
     }
 
+    @Test(groups = "certificate", priority = 1)
+    public void testAddApplicationWithCertificate() throws CertificateMgtException {
+
+        ServiceProvider inputSP = new ServiceProvider();
+        inputSP.setApplicationName(APPLICATION_NAME_1);
+        inputSP.setCertificateContent(certificate.getCertificateContent());
+        addApplicationConfigurations(inputSP);
+
+        reset(applicationCertificateManagementService);
+        doReturn(0).when(applicationCertificateManagementService).addCertificate(any(), anyString());
+        doReturn(certificate).when(applicationCertificateManagementService).getCertificateByName(any(), anyString());
+        try {
+            appIdWithCert = applicationManagementService.createApplication(inputSP, SUPER_TENANT_DOMAIN_NAME,
+                    REGISTRY_SYSTEM_USERNAME);
+        } catch (Exception e) {
+            Assert.fail();
+        }
```

Comment:
Ideally we should have asserted that the cert is added. Is this verified later

Reply:
Since the `ApplicationCertificateManagementService` is mocked, we cannot assert that the certificate is added. But as we added unit tests for the certificate-mgt component we can assure that the certificate addition should be successful.

Reply:
noted the test below

Reply:
If the cert is added successfully, certificateID, should be available in the spProperties array in the Service Provider object. That was asserted in the `testGetApplicationWithCertificate()` test method